install_config/syncing_groups_with_ldap.adoc

```diff
@@ -860,3 +867,277 @@ users: <5>
 ====
+
+[[recursive-sync]]
+== Recursive Membership Sync Example
+
+Groups in OpenShift do not nest.  Thus the LDAP server must flatten group membership before
+the data can be consumed.  Microsoft's Active Directory Server supports this feature via the
+https://msdn.microsoft.com/en-us/library/aa746475(v=vs.85).aspx[LDAP_MATCHING_RULE_IN_CHAIN]
+which has the OID `1.2.840.113556.1.4.1941`.  Furthermore, only explicitly
+xref:../install_config/syncing_groups_with_ldap.adoc#running-ldap-sync[whitelisted] groups
+can be synced when using this matching rule.
+
+This section has an example for the augmented Active Directory schema which synchronizes a
+group named *admins* that has one user *Jane* and one group *otheradmins* as members.
+The *otheradmins* group has one user member: *Jim*.  This example explains:
+
+* How the group and users are added to the LDAP server.
+* What the LDAP sync configuration file looks like.
+* What the resulting Group record in {product-title} will be after synchronization.
+
+In the augmented Active Directory schema, both users (Jane and Jim) and groups
+exist in the LDAP server as first-class entries, and group membership is stored
+in attributes on the user or the group. The following snippet of `ldif` defines the
+users and group for this schema:
+
+.LDAP Entries Using Augmented Active Directory Schema With Recursive Members: *_augmented_active_directory_recursive.ldif_*
+====
+[source,ldif]
+----
+dn: ou=users,dc=example,dc=com
+objectClass: organizationalUnit
+ou: users
+
+dn: cn=Jane,ou=users,dc=example,dc=com
+objectClass: person
+objectClass: organizationalPerson
+objectClass: inetOrgPerson
+objectClass: testPerson
+cn: Jane
+sn: Smith
+displayName: Jane Smith
+mail: jane.smith@example.com
+memberOf: cn=admins,ou=groups,dc=example,dc=com <1>
+
+dn: cn=Jim,ou=users,dc=example,dc=com
+objectClass: person
+objectClass: organizationalPerson
+objectClass: inetOrgPerson
+objectClass: testPerson
+cn: Jim
+sn: Adams
+displayName: Jim Adams
+mail: jim.adams@example.com
+memberOf: cn=otheradmins,ou=groups,dc=example,dc=com <1>
+
+dn: ou=groups,dc=example,dc=com
+objectClass: organizationalUnit
+ou: groups
+
+dn: cn=admins,ou=groups,dc=example,dc=com <2>
+objectClass: group
+cn: admins
+owner: cn=admin,dc=example,dc=com
+description: System Administrators
+member: cn=Jane,ou=users,dc=example,dc=com
+member: cn=otheradmins,ou=groups,dc=example,dc=com
+
+dn: cn=otheradmins,ou=groups,dc=example,dc=com <2>
+objectClass: group
+cn: otheradmins
+owner: cn=admin,dc=example,dc=com
+description: Other System Administrators
+memberOf: cn=admins,ou=groups,dc=example,dc=com <1> <3>
+member: cn=Jim,ou=users,dc=example,dc=com
+----
+<1> The user's and group's memberships are listed as attributes on the object.
+<2> The groups are first-class entries on the LDAP server.
+<3> The *otheradmins* group is a member of the *admins* group.
+====
+
+To sync this group, you must first create the configuration file. The augmented
```

Comment:
To sync nested groups with Active Directory, you must provide an LDAP query definition...